### PR TITLE
Add an option to configure the exporter buffer of the BatchProcessor

### PR DESCRIPTION
**Pull request by Frapschen**
resolve: https://github.com/open-telemetry/opentelemetry-go/issues/5238

### CODE REVIEW

File: sdk/log/batch.go [hunk 1]
```diff
@@ -19,11 +19,13 @@ const (
 	dfltExpInterval     = time.Second
 	dfltExpTimeout      = 30 * time.Second
 	dfltExpMaxBatchSize = 512
+	dfltExpBufferSize   = 1
 
 	envarMaxQSize        = "OTEL_BLRP_MAX_QUEUE_SIZE"
 	envarExpInterval     = "OTEL_BLRP_SCHEDULE_DELAY"
 	envarExpTimeout      = "OTEL_BLRP_EXPORT_TIMEOUT"
 	envarExpMaxBatchSize = "OTEL_BLRP_MAX_EXPORT_BATCH_SIZE"
+	envarExpBufferSize   = "OTEL_BLRP_EXPORT_BUFFER_SIZE"
```
Comment (MrAlias) [inline sdk/log/batch.go:28]: This option is not [a specification defined environment variable](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md#batch-logrecord-processor). [It should not be defined this way](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md#language-specific-environment-variables), however, I would prefer to not use environment variables.

File: sdk/log/batch.go [hunk 2]
```diff
@@ -458,3 +463,19 @@ func WithExportMaxBatchSize(size int) BatchProcessorOption {
 		return cfg
 	})
 }
+
+// WithExportBufferSize sets the buffer size of batch in every export.
+// Batches will be temporarily kept in a memory buffer until it exported successfully.
+//
+// If the OTEL_BLRP_EXPORT_BUFFER_SIZE environment variable is set,
+// and this option is not passed, that variable value will be used.
```
Comment (MrAlias) [inline sdk/log/batch.go:471]:
```suggestion
```

File: sdk/log/batch.go [hunk 3]
```diff
@@ -458,3 +463,19 @@ func WithExportMaxBatchSize(size int) BatchProcessorOption {
...
+//
+// By default, if an environment variable is not set, and this option is not
+// passed, 1 will be used.
```
Comment (MrAlias) [inline sdk/log/batch.go:474]:
```suggestion
// By default a size of 1 will be used.
```

File: sdk/log/batch.go [hunk 4]
```diff
@@ -458,3 +463,19 @@ func WithExportMaxBatchSize(size int) BatchProcessorOption {
 		return cfg
 	})
 }
+
+// WithExportBufferSize sets the buffer size of batch in every export.
```
Comment (MrAlias) [inline sdk/log/batch.go:467]:
```suggestion
// WithExportBufferSize sets the batch buffer size.
```

File: sdk/log/batch.go [hunk 5]
```diff
@@ -458,3 +463,15 @@ func WithExportMaxBatchSize(size int) BatchProcessorOption {
 		return cfg
 	})
 }
+
+// WithExportBufferSize sets the batch buffer size.
+// Batches will be temporarily kept in a memory buffer until it exported successfully.
```
Comment (dmathieu) [inline sdk/log/batch.go:468]:
```suggestion
// Batches will be temporarily kept in a memory buffer until they are exported.
```

File: sdk/log/batch.go [hunk 6]
```diff
@@ -458,3 +463,15 @@ func WithExportMaxBatchSize(size int) BatchProcessorOption {
...
+//
+// By default a size of 1 will be used.
```
Comment (dmathieu) [inline sdk/log/batch.go:470]:
```suggestion
// By default, a value of 1 will be used.
```

**Review by dmathieu — COMMENTED** [review]
Left minor comments.

**Review by MrAlias — APPROVED** [review]

**Review by dmathieu — APPROVED** [review]

**Pull request merged**